simulation/positrondetector.cc

```diff
@@ -78,4 +108,41 @@ G4bool MyPositronDetector::ProcessHits(G4Step *step, G4TouchableHistory *ROhist)
 }
 
 void MyPositronDetector::EndOfEvent(G4HCofThisEvent* hce)
-{}
+{
+    /*
+    // Prints based on the particles detected
+    if (photonsDetected && positronsDetected) {
+        G4cout << "Event with both photons and positrons" << G4endl;
+    } else if (photonsDetected) {
+        G4cout << "Event with only photons" << G4endl;
+    } else if (positronsDetected) {
+        G4cout << "Event with only positrons" << G4endl;
+    } else {
+        G4cout << "Event with no photons or positrons" << G4endl;
+    }
+    */
+
+    G4int eventNumber = G4RunManager::GetRunManager()->GetCurrentEvent()->GetEventID();
+    auto *man = G4RootAnalysisManager::Instance();
+
+    // Get the number of hits in the hitsCollection
+    G4int nOfHits = hitsCollection->entries();
+    // loop through the hitsCollection
+
+    for (G4int i = 0; i < nOfHits; i++) {
+        (*hitsCollection)[i]->Print();
+
+        G4int detectorNum = (*hitsCollection)[i]->GetNumberOfEvent();
+        G4int numberOfPhotons = (*hitsCollection)[i]->GetNumberOfPhotons();
+        G4int numberOfPositrons = (*hitsCollection)[i]->GetNumberOfPositrons();
+
+        man->FillNtupleIColumn(4, 0, eventNumber);
```

Comment:
move IFs outside

Reply:
By making some adjustments in the code I managed to fix the way the particles were being counted, for instance the plot for the number of positrons is:

![image](https://github.com/GASP-UFRGS/DarkPhoton-LNLS/assets/78316565/00b3a5a7-5459-4913-80a2-a587301b549a)

Considering that the events with no positrons detected are very few, It was necessary to zoom in the plot:

![image](https://github.com/GASP-UFRGS/DarkPhoton-LNLS/assets/78316565/cfcccc8b-fc13-4345-92c0-771f57eea947)

Alternatively, we can also see the same behaviour with the photons, where it was also necessary to zoom in the graph a little:

![image](https://github.com/GASP-UFRGS/DarkPhoton-LNLS/assets/78316565/61c95f20-6b76-4988-aeaa-39a6471085b9)

![image](https://github.com/GASP-UFRGS/DarkPhoton-LNLS/assets/78316565/9f50799e-923d-4f11-9168-dc2b7f87360f)